src/OrchardCore/OrchardCore.Workflows.Abstractions/Models/WorkflowExecutionContext.cs

```diff
@@ -0,0 +1,150 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using OrchardCore.Scripting;
+using OrchardCore.Workflows.Services;
+
+namespace OrchardCore.Workflows.Models
+{
+    public class WorkflowExecutionContext
+    {
+        private readonly IEnumerable<IWorkflowExecutionContextHandler> _handlers;
+        private readonly IWorkflowExpressionEvaluator _expressionEvaluator;
+        private readonly IWorkflowScriptEvaluator _scriptEvaluator;
+        private readonly IServiceProvider _serviceProvider;
+        private readonly ILogger<WorkflowExecutionContext> _logger;
+
+        public WorkflowExecutionContext
+        (
+            WorkflowDefinition workflowDefinitionRecord,
+            WorkflowInstance workflowInstanceRecord,
+            IServiceProvider serviceProvider,
+            IDictionary<string, object> input,
+            IDictionary<string, object> output,
+            IDictionary<string, object> properties,
+            IList<ExecutedActivity> executedActivities,
+            object lastResult,
+            IEnumerable<ActivityContext> activities,
+            IEnumerable<IWorkflowExecutionContextHandler> handlers,
+            IWorkflowExpressionEvaluator expressionEvaluator,
```

Comment:
Remove from here and let each activity resolve the services they need.

Reply:
Done.